app/src/main/java/pl/gittobefit/WorkoutDisplay/fragments/DisplayReceivedTraining.java

```diff
@@ -51,14 +67,47 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         model.getPosition().observe(getViewLifecycleOwner(), integer -> index = integer);
     }
 
+
     @Override
     public void onResume() {
 
         super.onResume();
+        List<SavedTraining> result2 = AppDataBase.getInstance(getContext()).training().getInfoForTrainingList();
+
+        if (index != UserTrainings.getInstance().getTrainingArrayList().size() && UserTrainings.getInstance().getTraining(index).getPlanList().size() == 0) {
+            SavedTraining savedTraining = AppDataBase.getInstance(getContext()).training().getOneTraining(result2.get(index).getId());
```

Comment:
Jeżeli dobrze to rozumiem, to w liście ```result2``` powinieneś mieć ten trening i wtedy nie trzeba by było pobierać go z bazy